ddtrace/tracer/span.go

```diff
@@ -358,19 +359,23 @@ func (s *span) finish(finishTime int64) {
 		}
 		if feats.DropP0s {
 			// the agent supports dropping p0's in the client
-			if shouldDrop(s) {
+			keep = shouldKeep(s)
+			if keep {
 				// ...and this span can be dropped
 				atomic.AddUint64(&t.droppedP0Spans, 1)
 				if s == s.context.trace.root {
 					atomic.AddUint64(&t.droppedP0Traces, 1)
 				}
-				return
 			}
 		}
 	}
 	if s.context.drop {
-		// not sampled by local sampler
-		return
+		// client sampling enabled. This will lead to inexact APM metrics but reduces performance impact.
+		keep = false
```

Comment:
The change of logic here makes me wonder whether we can have just one of the two. Can we just have `keep` instead of both `keep` and `drop`? The `drop` field is currently scoped to the `spanContext` even though the local rate sampler samples based on TraceID, so in practice all spans in that trace would be dropped, just like `keep` works.